WorkloadManagementSystem/Client/Matcher.py

```diff
@@ -0,0 +1,329 @@
+
+      for k in ( 'DIRACVersion', 'ReleaseVersion', 'ReleaseProject', 'VirtualOrganization' ):
+        if classAdAgent.lookupAttribute( k ):
+          resourceDict[ k ] = classAdAgent.getAttributeString( k )
+
+    else:
+      for name in singleValueDefFields:
+        if resourceDescription.has_key( name ):
+          resourceDict[name] = resourceDescription[name]
+
+      for name in multiValueMatchFields:
+        if resourceDescription.has_key( name ):
+          resourceDict[name] = resourceDescription[name]
+
+      if resourceDescription.has_key( 'JobID' ):
+        resourceDict['JobID'] = resourceDescription['JobID']
+
+      for k in ( 'DIRACVersion', 'ReleaseVersion', 'ReleaseProject', 'VirtualOrganization', 'PilotReference', 'PilotBenchmark' ):
+        if k in resourceDescription:
+          resourceDict[ k ] = resourceDescription[ k ]
+
+    return resourceDict
+
+
+
+  def _reportStatus( self, resourceDict, jobID ):
+    """ Reports the status of the matched job in jobDB and jobLoggingDB
+
+        Do not fail if errors happen here
+    """
+    attNames = ['Status', 'MinorStatus', 'ApplicationStatus', 'Site']
+    attValues = ['Matched', 'Assigned', 'Unknown', resourceDict['Site']]
+    result = self.jobDB.setJobAttributes( jobID, attNames, attValues )
+    if not result['OK']:
+      self.log.error( "Problem reporting job status", "setJobAttributes, jobID = %s: %s" % ( jobID, result['Message'] ) )
+    else:
+      self.log.verbose( "Set job attributes for jobID %s" % jobID )
+
+    result = self.jlDB.addLoggingRecord( jobID,
+                                         status = 'Matched',
+                                         minor = 'Assigned',
+                                         source = 'Matcher' )
+    if not result['OK']:
+      self.log.error( "Problem reporting job status", "addLoggingRecord, jobID = %s: %s" % ( jobID, result['Message'] ) )
+    else:
+      self.log.verbose( "Added logging record for jobID %s" % jobID )
+
+
+  def _checkMask( self, resourceDict ):
+    """ Check the mask: are we allowed to run normal jobs?
+
+        FIXME: should we move to site OR SE?
+    """
+    if not 'Site' in resourceDict:
+      self.log.error( "Missing Site Name in Resource JDL" )
+      raise RuntimeError( "Missing Site Name in Resource JDL" )
+
+    # Get common site mask and check the agent site
+    result = self.jobDB.getSiteMask( siteState = 'Active' )
+    if not result['OK']:
+      self.log.error( "Internal error", "getSiteMask: %s" % result['Message'] )
+      raise RuntimeError( "Internal error" )
+    maskList = result['Value']
+
+    if resourceDict['Site'] not in maskList:
+      return False
+
+    return True
+
+  def _updatePilotInfo( self, resourceDict, jobID ):
+    """ Update pilot information - do not fail if we don't manage to do it
+
+        FIXME:  this part should not be done here, by the matcher. Instead, it should be done by the pilot itself.
```

Comment:
PilotInfoReportedFlag is there in order not to enter the pilot info into the PilotAgentsDB each time the pilot is contacting the Matcher. The matcher does not know if the pilot info is in the DB, but the pilot knows.